### PR TITLE
Support for `UserAssignedIdentity` in `azurerm_storage_account`

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/11751

--- PASS: TestAccStorageAccount_systemAssignedIdentity (160.00s)
--- PASS: TestAccStorageAccount_userAssignedIdentity (171.82s)
--- PASS: TestAccStorageAccount_systemAssignedUserAssignedIdentity (171.53s)
--- PASS: TestAccStorageAccount_updateResourceByEnablingIdentity (421.15s)

### CODE REVIEW

File: azurerm/internal/services/storage/storage_account_resource.go [hunk 1]
```diff
@@ -314,7 +316,6 @@ func resourceStorageAccount() *pluginsdk.Resource {
 			"identity": {
 				Type:     pluginsdk.TypeList,
 				Optional: true,
-				Computed: true,
```
Comment (katbyte) [inline azurerm/internal/services/storage/storage_account_resource.go:317]: how come we are removing this? and will it affect people expecting this to be populated?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: azurerm/internal/services/storage/storage_account_resource.go (hunk 1 above)
Comment (yupwei68) [reply azurerm/internal/services/storage/storage_account_resource.go:317]: It was not `Computed`, if not set, it'll not return a default value. It causes the problem that if the user remove the block `identity`, it'll not detect the change.
But I could keep it as origin, if it breaks.

**Comment by yupwei68** [comment]
Thanks kt for your comments. 
I've updated according to your comment. But on fact, I still think making `identity` `Computed` prevents users from disabling `identity`. 
All tests have passed.
Please continue reviewing.

**Review by katbyte — APPROVED** [review]
Thanks @yupwei68 - LGTM 👍

**Pull request merged**